src/lib.rs

```diff
@@ -185,52 +189,163 @@ fn enforce_container_security_policies(
 fn validate(payload: &[u8]) -> CallResult {
     info!(LOG_DRAIN, "starting validation");
```

Comment:
nit, I know this is part of the template. It doesn't look like we are using logging, hence I would just remove all that stuff.